pkgs/applications/networking/instant-messengers/quaternion/default.nix

```diff
@@ -11,20 +11,30 @@ stdenv.mkDerivation rec {
     sha256 = "0zrr4khbbdf5ziq65gi0cb1yb1d0y5rv18wld22w1x96f7fkmrib";
   };
 
-  buildInputs = [ qtbase qtquickcontrols libqmatrixclient ];
+  buildInputs = [ qtbase qtquickcontrols ];
 
   nativeBuildInputs = [ cmake ];
 
+  patches = [
+    # https://github.com/QMatrixClient/Quaternion/pull/400
+    (fetchpatch {
+      url = "https://github.com/QMatrixClient/Quaternion/commit/6cb29834efc343dc2bcf1db62cfad2dc4c121c54.patch";
+      sha256 = "0n7mgzzrvx9sa657rfb99i0mjh1k0sn5br344mknqy3wgqdr7s3x";
+    })
+  ];
+
   # libqmatrixclient is now compiled as a dynamic library but quarternion cannot use it yet
   # https://github.com/QMatrixClient/Quaternion/issues/239
   postPatch = ''
     rm -rf lib
     ln -s ${libqmatrixclient.src} lib
   '';
 
-  postInstall = ''
+  postInstall = lib.optionalString (!stdenv.isDarwin) ''
     substituteInPlace $out/share/applications/quaternion.desktop \
       --replace 'Exec=quaternion' "Exec=$out/bin/quaternion"
+  '' + lib.optionalString stdenv.isDarwin ''
+    mv $out/bin $out/Applications
```

Comment:
Is the application the only thing in bin? It's probably better to only move the application in case anything else ever gets added.